pi/heartbeat.py

```diff
@@ -147,6 +147,28 @@ def send_heartbeat(flic_last_seen_secs, flic_last_ping_secs, system_id):
         logging.warning("error sending heartbeat", exc_info=e)
         return False
 
+
+last_ping = datetime.datetime.now()
+
+def gather_stats():
+    html = get_darkstat_html()
+    ip = parse_flic_ip_from_darkstat_html(html, FLIC_MAC_ADDRESS)
+    if ping(ip):
+        last_ping = datetime.datetime.now()
+    num_secs_darkstat = parse_flic_last_seen_from_darkstat_html(html, FLIC_MAC_ADDRESS)
+    num_secs_ping = (datetime.datetime.now() - last_ping).total_seconds()
+    return (num_secs_darkstat, num_secs_ping)
+
+# run the loop at most once every N seconds (N = min_delay)
+# if the run loop takes longer then N seconds we don't need to sleep
+def run_loop_delay(min_delay, last_run_time):
+
```

Comment:
Feels like this newline would be better off removed. Otherwise, the function looks a little funny. Your call.

Reply:
done